linchpin/shell/__init__.py

```diff
@@ -227,11 +230,18 @@ def up(ctx, targets, run_id, tx_id, inventory_format):
     run-id:     Use the data from the provided run_id value
     """
 
+    flags = {}
+    if run_hooks_on_failure:
+        flags["run_hooks_on_failure"] = run_hooks_on_failure
+    if no_hooks:
+        flags["no_hooks"] = no_hooks
+
```

Comment:
I kind of gave an example elsewhere of how to do this, but here's a simple way to get this data into context (or update it if it's changing) instead of using another variable (`flags` in this case).

`ctx.set_cfg('no_hooks', no_hooks)`
`self.ctx.set_cfg('run_hooks_on_failure', run_hooks_on_failure)`

When you need to create the flags object in the cli library, say, just get the data out there.